kombu/transport/virtual/exchange.py

```diff
@@ -75,8 +82,14 @@ class TopicExchange(ExchangeType):
     _compiled = {}
 
     def lookup(self, table, exchange, routing_key, default):
-        return [queue for rkey, pattern, queue in table
-                if self._match(pattern, routing_key)]
+        # Using an OrderedDict to purge queue duplicates while
+        # keeping the same order in which they appear in the table
+        return list(
```

Comment:
This is nice! I didn't know about .fromkeys.  You can probably remove `list(` here to avoid making a copy, as using a dict as a list will already use the keys.